lib/matplotlib/artist.py

```diff
@@ -523,6 +535,12 @@ def set_snap(self, snap):
             segments, round to the nearest pixel center
 
         Only supported by the Agg and MacOSX backends.
+
+        Parameters
+        ----------
+        snap : Optional[bool]
+            ..
+                ACCEPTS: Optional[bool]
```

Comment:
This renders to this
![image](https://user-images.githubusercontent.com/2743474/32464127-104b2a0a-c350-11e7-9573-7ecd960174e3.png)
Is it what you meant?
P.S: For me an optional argument is the one you can skip, so in that case `.set_snap()` should be valid, but here we have `[ None | bool ]`

Reply:
No, Optional[t] means Union[t, None] (https://docs.python.org/3/library/typing.html#typing.Optional).

Reply:
Well, that was confusing, thanks for clarifying.

Reply:
Actually I agree that `foo or None` is less confusing than `Optional[None]`, fixed accordingly.